src/libs/OptionsListUtils.js

```diff
@@ -54,14 +54,24 @@ Onyx.connect({
     },
 });
 
+const expenseReports = {};
 const iouReports = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
-    callback: (iouReport, key) => {
-        if (!iouReport || !key || !iouReport.ownerEmail || !ReportUtils.isIOUReport(iouReport)) {
+    callback: (report, key) => {
+        if (!report || !key || !report.ownerEmail) {
             return;
         }
-        iouReports[key] = iouReport;
+
+        if (ReportUtils.isExpenseReport(report)) {
+            expenseReports[key] = report;
+            return;
+        }
+
+        if (!ReportUtils.isIOUReport(report)) {
+            return;
+        }
+        iouReports[key] = report;
```

Comment:
> This section feels weird to me though, why not just do the same as you did for isExpenseReport instead of having the early return for not IOUs? I guess it would be slightly faster but seems more confusing as a reader.

I agree. The reason I originally wrote it that way is we've got this custom `prefer-early-return` lint rule that I thought would complain, but I guess not so I updated this to make it more readable.